components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtValidator.java

```diff
@@ -381,6 +383,17 @@ private void validateClaimsConfigs(List<String> validationMsg, ClaimConfig claim
                     subjectClaimUri)) {
                 validationMsg.add(String.format(CLAIM_NOT_AVAILABLE, subjectClaimUri, tenantDomain));
             }
+        } else {
+            if (StringUtils.isNotBlank(roleClaimUri) && !(allRemoteClaimUris).contains(roleClaimUri)) {
+                validationMsg.add(String.format(CLAIM_NOT_AVAILABLE, roleClaimUri, tenantDomain));
```

Comment:
We might need to use a different error message here.
"Local claim %s is not available in the server for tenantDomain:%s". Might not be suitable here.

Reply:
Is it okay to have "Service Provider claim %s is not available for Service Provider:%s." error message. 
"String serviceProviderName" will be added to **validateClaimsConfigs** method signature

Reply:
Application Claim URI: xyz is not defined for Service Provider : %s 
or something along the lines would be fine :)

Reply:
addressed in 5f538eb3b5513b3978a784dd8a0df158de48ae34